### PR TITLE
Kernel nystroem approximation update with list of landmarks.

### PR DESCRIPTION
**Pull request by hurricane642**
Hello everyone! Here is the code we would like to push to `scikit-learn`. The change we've added is that it's now possible to pass a list of landmarks to kernel_approximation.py. Any suggestions to improve the code/documentation/tests are welcome.

### CODE REVIEW

File: sklearn/kernel_approximation.py [hunk 1]
```diff
@@ -781,8 +791,7 @@ def fit(self, X, y=None):
         U, S, V = svd(basis_kernel)
         S = np.maximum(S, 1e-12)
         self.normalization_ = np.dot(U / np.sqrt(S), V)
-        self.components_ = basis
-        self.component_indices_ = inds
+        self.components_ = X[self.component_indices_]
```
Comment (rosecers) [inline sklearn/kernel_approximation.py:794]: Why is this written like this? 
1) self.component_indices_ does not look to be set, and 
2) basis still works as the self.components_
```suggestion
        self.components_ = basis
        self.component_indices_ = basis_inds
```

File: sklearn/kernel_approximation.py [hunk 2]
```diff
@@ -768,9 +769,18 @@ def fit(self, X, y=None):
...
         n_components = min(n_samples, n_components)
-        inds = rnd.permutation(n_samples)
-        basis_inds = inds[:n_components]
-        basis = X[basis_inds]
+
+        if landmarks is None:
+            inds = rnd.permutation(n_samples)
+            basis_inds = inds[:n_components]
+        else:
+            basis_inds = column_or_1d(landmarks)
```
Comment (rosecers) [inline sklearn/kernel_approximation.py:777]: Do the validation of basis_inds here, not below

**Review by rosecers — CHANGES_REQUESTED** [review]
Hey @hurricane642 and @bhelfrecht. This is almost there, but still needs work. You should create a new PR that contains the requested changes and a better PR message.

Still remaining:
- You need to add documentation to `doc/modules/kernel_approximation.rst`
- All CI must pass
- your tests need better explanation and documentation. 
- The PR itself needs to reference the issue it addresses (see the scikit-cosmo notes for this)
- Including the scikit-learn checklist is also very useful in the PR to show that you have taken everything into account(https://scikit-learn.org/stable/developers/contributing.html, see below)

- [ ] Give your pull request a helpful title that summarises what your contribution does. This title will often become the commit message once merged so it should summarise your contribution for posterity. In some cases “Fix <ISSUE TITLE>” is enough. “Fix #<ISSUE NUMBER>” is never a good title.
- [ ] Make sure your code passes the tests. The whole test suite can be run with pytest, but it is usually not recommended since it takes a long time. It is often enough to only run the test related to your changes: for example, if you changed something in sklearn/linear_model/logistic.py, running the following commands will usually be enough:
    - [ ] pytest sklearn/linear_model/logistic.py to make sure the doctest examples are correct
    - [ ] pytest sklearn/linear_model/tests/test_logistic.py to run the tests specific to the file
    - [ ] pytest sklearn/linear_model to test the whole linear_model module
    - [ ] pytest doc/modules/linear_model.rst to make sure the user guide examples are correct.
    - [ ] pytest sklearn/tests/test_common.py -k LogisticRegression to run all our estimator checks (specifically for LogisticRegression, if that’s the estimator you changed).
- [ ] Make sure your code is properly commented and documented, and make sure the documentation renders properly. To build the documentation, please refer to our Documentation guidelines. The CI will also build the docs: please refer to Generated documentation on CircleCI.
- [ ] Tests are necessary for enhancements to be accepted. Bug-fixes or new features should be provided with non-regression tests. These tests verify the correct behavior of the fix or feature. In this manner, further modifications on the code base are granted to be consistent with the desired behavior. In the case of bug fixes, at the time of the PR, the non-regression tests should fail for the code base in the main branch and pass for the PR code.
- [ ] Make sure that your PR does not add PEP8 violations. To check the code that you changed, you can run the following command (see above to set up the upstream remote): `git diff upstream/main -u -- "*.py" | flake8 --diff` or `make flake8-diff` which should work on unix-like system.
- [ ] Follow the [Coding guidelines](https://scikit-learn.org/stable/developers/develop.html#coding-guidelines).
- [ ] When applicable, use the validation tools and scripts in the sklearn.utils submodule. A list of utility routines available for developers can be found in the Utilities for Developers page.
- [ ] Often pull requests resolve one or more other issues (or pull requests). If merging your pull request means that some other issues/PRs should be closed, you should use keywords to create link to them (e.g., Fixes #1234; multiple issues/PRs are allowed as long as each one is preceded by a keyword). Upon merging, those issues/PRs will automatically be closed by GitHub. If your pull request is simply related to some other issues/PRs, create a link to them without using the keywords (e.g., See also #1234).
- [ ] PRs should often substantiate the change, through benchmarks of performance and efficiency (see Monitoring performance) or through examples of usage. Examples also illustrate the features and intricacies of the library to users. Have a look at other examples in the examples/ directory for reference. Examples should demonstrate why the new functionality is useful in practice and, if possible, compare it to other methods available in scikit-learn.
- [ ] New features often need to be illustrated with narrative documentation in the user guide, with small code snippets. If relevant, please also add references in the literature, with PDF links when possible.
- [ ] The user guide should also include expected time and space complexity of the algorithm and scalability, e.g. “this algorithm can scale to a large number of samples > 100000, but does not scale in dimensionality: n_features is expected to be lower than 100”.

**Comment by hurricane642** [comment]
So, since I have a feeling that I have errors going on somewhere already in the main, I will close this PR and fork and start a new one. I'll move your comments carefully :).

**Pull request closed without merging**